### ISSUE TITLE
error in align_cds

### ISSUE THREAD

**Issue by JingjingSherryWu** [cole-trapnell-lab/monocle3#337]
** If this is a question and not a bug report or enhancement request, please post to our google group at https://groups.google.com/forum/#!forum/monocle-3-users**

**Describe the bug**
Error 'number of rows is not the same across batches' when running align_cds using the demo dataset.

**To Reproduce**
I ran the exact code as on the tutorial, as following:
expression_matrix <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_expression.rds"))
cell_metadata <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_colData.rds"))
gene_annotation <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_rowData.rds"))
cds <- new_cell_data_set(expression_matrix, cell_metadata = cell_metadata, gene_metadata = gene_annotation)
cds <- preprocess_cds(cds, num_dim = 100)
cds = align_cds(cds, num_dim = 100, alignment_group = "plate")

**traceback()**
4: stop(sprintf("number of %ss is not the same across batches", 
       DIM))
3: checkBatchConsistency(batches, cells.in.columns = TRUE)
2: batchelor::fastMNN(as.matrix(preproc_res), batch = colData(cds)[, 
       alignment_group], k = alignment_k, cos.norm = FALSE, pc.input = TRUE)
1: align_cds(cds, num_dim = 100, alignment_group = "plate")

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/42477063/81141091-883f9e80-8f39-11ea-9b9b-295364163279.png)

**sessionInfo():**
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] dplyr_0.8.5                 monocle3_0.2.1              SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1 DelayedArray_0.14.0        
 [6] matrixStats_0.56.0          GenomicRanges_1.40.0        GenomeInfoDb_1.24.0         IRanges_2.22.1              S4Vectors_0.26.0           
[11] Biobase_2.48.0              BiocGenerics_0.34.0        

loaded via a namespace (and not attached):
 [1] viridis_0.5.1             pkgload_1.0.2             BiocSingular_1.4.0        viridisLite_0.3.0         DelayedMatrixStats_1.10.0 assertthat_0.2.1         
 [7] BiocManager_1.30.10       vipor_0.4.5               GenomeInfoDbData_1.2.3    remotes_2.1.1             sessioninfo_1.1.1         pillar_1.4.4             
[13] backports_1.1.6           lattice_0.20-41           glue_1.4.0                digest_0.6.25             XVector_0.28.0            colorspace_1.4-1         
[19] Matrix_1.2-18             plyr_1.8.6                pkgconfig_2.0.3           devtools_2.3.0            zlibbioc_1.34.0           purrr_0.3.4              
[25] scales_1.1.0              processx_3.4.2            batchelor_1.4.0           BiocParallel_1.22.0       tibble_3.0.1              ggplot2_3.3.0            
[31] usethis_1.6.1             ellipsis_0.3.0            withr_2.2.0               cli_2.0.2                 magrittr_1.5              crayon_1.3.4             
[37] memoise_1.1.0             ps_1.3.2                  fs_1.4.1                  fansi_0.4.1               beeswarm_0.2.3            pkgbuild_1.0.7           
[43] scater_1.16.0             tools_4.0.0               prettyunits_1.1.1         lifecycle_0.2.0           stringr_1.4.0             munsell_0.5.0            
[49] irlba_2.3.3               callr_3.4.3               compiler_4.0.0            rsvd_1.0.3                rlang_0.4.6               grid_4.0.0               
[55] RCurl_1.98-1.2            BiocNeighbors_1.6.0       rstudioapi_0.11           igraph_1.2.5              bitops_1.0-6              testthat_2.3.2           
[61] gtable_0.3.0              curl_4.3                  reshape2_1.4.4            R6_2.4.1                  gridExtra_2.3             rprojroot_1.3-2          
[67] desc_1.2.0                ggbeeswarm_0.6.0          stringi_1.4.6             Rcpp_1.0.4.6              vctrs_0.2.4               tidyselect_1.0.0         

**Additional context**

**Comment by kh49** [comment]
I had the same issue. It looks like alignment.R is passing the argument pc.input = TRUE which isn't a named argument in fastMNN to the fastMNN function. I was able to get it working by removing that line and transposing the input matrix containing the PCs.

```
preproc_res <- reducedDims(cds)[["PCA"]]

corrected_PCA = batchelor::fastMNN(as.matrix(t(preproc_res)),
                                   batch=colData(cds)[,"plate"],
                                   k=50,
                                   cos.norm=FALSE
                            )

```

**Comment by akpgreentree** [comment]
I've also seen this issue. It looks like it has to do with the latest version of batchelor (1.4.0), which is automatically installed by Bioconductor version 3.11, in R 4.0.0.

It seems like this isn't *quite* a bug, as the installation page for monocle3 specifies having bioconductor version 3.10; but it's certainly not ideal.

**Comment by oagray** [comment]
> I've also seen this issue. It looks like it has to do with the latest version of batchelor (1.4.0), which is automatically installed by Bioconductor version 3.11, in R 4.0.0.
> 
> It seems like this isn't _quite_ a bug, as the installation page for monocle3 specifies having bioconductor version 3.10; but it's certainly not ideal.

I'm having the same issue with both my data on the test data. I've tried reinstalling batchelor 1.3.6, but I get the same error. Is there a workaround for this apart from having an earlier version of R?

**Comment by brgew** [comment]
Hi,

I believe that this problem is fixed in the Monocle3 develop branch. You can install it using the command

`devtools::install_github("https://github.com/cole-trapnell-lab/monocle3", ref="develop")`

Please let me know if the issue persists.

Thank you.

**Comment by hpliner** [comment]
I believe fixed in 0.2.2
